components/action-mgt/org.wso2.carbon.identity.action.management/src/test/java/org/wso2/carbon/identity/action/management/dao/ActionManagementDAOImplTest.java

```diff
@@ -0,0 +1,341 @@
+    private MockedStatic<IdentityTenantUtil> identityTenantUtil;
+    private MockedStatic<IdentityUtil> identityUtil;
+    private static final String DB_NAME = "action_mgt";
+    private ActionSecretProcessor actionSecretProcessor;
+    private SecretManagerImpl secretManager;
+
+    private final AuthType authType = new AuthType.AuthTypeBuilder()
+            .type(AuthType.AuthenticationType.BEARER)
+            .properties(Arrays.asList(
+                    new AuthProperty.AuthPropertyBuilder().name("accesstoken").value("123456").isConfidential(true)
+                            .build()))
+            .build();
+
+    @BeforeClass
+    public void setUpClass() throws Exception{
+        initiateH2Database(getFilePath());
+    }
+
+    @BeforeMethod
+    public void setUp() {
+
+        daoImpl = new ActionManagementDAOImpl();
+        identityTenantUtil = mockStatic(IdentityTenantUtil.class);
+        identityDatabaseUtil = mockStatic(IdentityDatabaseUtil.class);
+        identityUtil = mockStatic(IdentityUtil.class);
+        actionSecretProcessor = mock(ActionSecretProcessor.class);
+        secretManager = mock(SecretManagerImpl.class);
+        ActionMgtServiceComponentHolder.getInstance().setSecretManager(secretManager);
+        try{
+            identityDatabaseUtil.when(() -> IdentityDatabaseUtil.getDBConnection(anyBoolean()))
+                        .thenReturn(getConnection());
+            identityTenantUtil.when(()-> IdentityTenantUtil.getTenantId(anyString())).thenReturn(TENANT_ID);
+            SecretType secretType = mock(SecretType.class);
+            when(secretType.getId()).thenReturn("secretId");
+            when(secretManager.getSecretType(any())).thenReturn(secretType);
+        }
+        catch (Exception ignored){
+        }
+    }
+
+    @AfterMethod
+    public void tearDown() {
+        identityTenantUtil.close();
+        identityDatabaseUtil.close();
+        identityUtil.close();
+    }
+
+    @AfterClass
+    public void wrapUp() throws Exception {
+        closeH2Database();
+    }
+
+    @Test
+    public void testAddAction() throws ActionMgtException {
+
+        Action action = getAction("120120");
+        Action result = daoImpl.addAction(action.getType().getActionType(), action.getId(), action, TENANT_ID);
+        Assert.assertEquals(action.getName(),result.getName());
+        Assert.assertEquals(action.getDescription(),result.getDescription());
+        Assert.assertEquals(action.getType(),result.getType());
+        Assert.assertEquals(action.getStatus(),result.getStatus());
+        Assert.assertEquals(action.getEndpoint().getUri(),result.getEndpoint().getUri());
+        Assert.assertEquals(action.getEndpoint().getAuthentication().getType(), result.getEndpoint().getAuthentication().getType());
+
+    }
+
+    @Test
+    public void testAddActionWithoutOptional() throws ActionMgtException {
+
+        Action action = getActionWithoutOptionals("136136");
```

Comment:
```suggestion
        Action action = getActionWithoutOptionals("5396d331-8abd-46d7-8841-0ef379d9c92e");
```
UUID is a 36 character long string. Hence lets add an example of a 36 character long string